rust-connectors/sinks/kafka/src/main.rs

```diff
@@ -18,15 +20,50 @@ async fn main() -> anyhow::Result<()> {
         println!("{}", serde_json::to_string_pretty(&schema).unwrap());
         return Ok(());
     }
-    let opts: KafkaOpt = KafkaOpt::from_args();
-    opts.common.enable_logging();
+    let raw_opts = KafkaOpt::from_args();
+    raw_opts.common.enable_logging();
     info!(
         connector_version = env!("CARGO_PKG_VERSION"),
         git_hash = git_hash_version(),
         "Starting Kafka sink connector",
     );
 
-    opts.execute().await?;
+    let kafka_sink_deps = KafkaSinkDependencies::from_kafka_opt(raw_opts).await?;
+    let kafka_producer = &kafka_sink_deps.kafka_producer;
+    let kafka_partition = &kafka_sink_deps.kafka_partition;
+    let kafka_topic = &kafka_sink_deps.kafka_topic;
+
+    info!("Starting stream");
+
+    let mut stream = kafka_sink_deps
+        .common_connector_opt
+        .create_consumer_stream()
+        .await?;
+
+    while let Some(Ok(record)) = stream.next().await {
+        let mut kafka_record = FutureRecord::to(kafka_topic.as_str())
+            .payload(record.value())
+            .key(record.key().unwrap_or(&[]));
+        if let Some(kafka_partition) = &(kafka_partition) {
+            kafka_record = kafka_record.partition(*kafka_partition);
+        }
+
+        let enqueue_res = kafka_producer.send_result(kafka_record);
+
+        if let Err((error, _)) = enqueue_res {
+            error!("KafkaError {:?}", error);
+
+            let last_record_read = serde_json::json!({
+                "offset": &record.offset,
+                "partition": &record.partition,
+            });
+            println!(
+                "{}",
+                serde_json::to_string(&last_record_read).unwrap_or("".into())
```

Comment:
We construct a `Value` just to serialize it again in the next line, this is not efficient. We could just format a string immediately. For example, `error!("KafkaError {:?}, offset: {}, partition: {}", error, &record.offset, &record.partition);`

Why do we need it in JSON format, btw? and why `println`?

Reply:
JSON line is machine-readable. It is useful for automation in the future. Is it unnecessary?

Reply:
For automation purposes, it's likely should be done for the entire program. Usually, it is configured at the logger level once, and no need to deal with it in every log printing.

Reply:
What I mean by automation is for other components that may consume the `stdout/stderr` (e.g when piped into something like kibana, these lines of output will be a bit easier to digest rather than an unstructured string, or scheduler that may have to manage the runtime of the connectors). 
> `println!`
My bad, it should have used something that puts to `stderr`

Either way, those are things out of the scope of this issue. I'll revert it back to using `error!`

Reply:
Yes, but for Kibana would be useful to get in a structured way not only the message itself but also a timestamp, module, log level, etc. 
Instead of: `2022-08-03T08:43:38Z INFO main some message` 
parse
`{
   "timestamp": "2022-08-03T08:43:38Z",
   "level": "INFO",
   "module": "main",
   "body": "main some message"
} 
`
right?
And that better to be done in one place. For example, using custom formatting like https://docs.rs/env_logger/0.9.0/env_logger/#using-a-custom-format, if there are no out-of-box solutions for JSON